### PR TITLE
feat: Timeline component

### PR DESCRIPTION
**Pull request by abhaywawale**
## Description
This is for timeline component for vuetify.

## Motivation and Context
see [#1340](https://github.com/vuetifyjs/vuetify/issues/1340)
## How Has This Been Tested?
In playground and jest and also visually too.

## Markup:
```vue
<v-timeline>
  <v-timeline-item icon="flight" icon-color="white" icon-fill-color="red" v-for="(n,i) in 5" :key="i">
    <template slot="item">
      <v-card flat>
        <v-card-media src="https://vuetifyjs.com/static/doc-images/cards/desert.jpg" height="200px">
        </v-card-media>
        <v-card-title primary-title>
          <div>
            <h3 class="headline mb-0">headline</h3>
            <div>description</div>
          </div>
        </v-card-title>
        <v-card-actions>
          <v-btn flat color="primary">text</v-btn>
          <v-btn flat color="primary">text</v-btn>
        </v-card-actions>
      </v-card>
    </template>
  </v-timeline-item>
</v-timeline>
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: yarn-error.log [hunk 1]
```diff
@@ -0,0 +1,7951 @@
+Arguments:
```
Comment (routegroup) [inline yarn-error.log]: It shouldn't be here.

**Comment by routegroup** [comment]
@abhaywawale `yarn-error.log` shouldn't be in repo.

File: dev/Playground.example.vue [hunk 2]
```diff
@@ -1,11 +0,0 @@
...
-  <v-app>
-  </v-app>
-</template>
-
-<script>
-  export default {
-    data: () => ({
-    })
-  }
-</script>
```
Comment (KaelWD) [inline dev/Playground.example.vue]: Don't change unrelated files

File: src/components/VTimeline/VTimeline.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      noIconParent: this.noIcon,
+      lineColorParent: this.lineColor,
+      hoverParent: this.hover,
+      raisedParent: this.raised,
+      circleFillColorParent: this.circleFillColor,
+      circleOutlineColorParent: this.circleOutlineColor,
+      lineSizeParent: this.lineSize,
+      circleOutlineSizeParent: this.circleOutlineSize,
+      hideCircleOutlineParent: this.hideCircleOutline
+    }
```
Comment (KaelWD) [inline src/components/VTimeline/VTimeline.js]: Are these reactive? I'd look into using registerable otherwise

File: src/components/VTimeline/VTimelineItem.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        'timeline--hover': this.hover || this.hoverParent,
+        'timeline--raised': this.raised || this.raisedParent
+      })
+    },
+    lineClasses () {
+      const lineColor = this.lineColor ? this.lineColor : this.lineColorParent
+      return this.addTextColorClassChecks({}, lineColor)
+    },
+    circleClasses () {
+      const circleFillColorClass = this.circleFillColor ? this.circleFillColor : this.circleFillColorParent
```
Comment (KaelWD) [inline src/components/VTimeline/VTimelineItem.js]: `this.circleFillColor || this.circleFillColorParent`

File: src/components/VTimeline/VTimelineItem.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+      const classes = {
+        [circleFillColorClass]: true
+      }
+      const circleOutlineColor = this.circleOutlineColor
+        ? this.circleOutlineColor
+        : this.circleOutlineColorParent ? this.circleOutlineColorParent : this.lineColor ? this.lineColor : this.lineColorParent
```
Comment (KaelWD) [inline src/components/VTimeline/VTimelineItem.js]: `this.circleOutlineColor || this.circleOutlineColorParent || this.lineColor || this.lineColorParent`

File: src/stylus/components/_timeline.styl [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    border-radius: 50%
+    background: inherit
+    box-shadow: currentColor 0px 0px 0px 8px 
+    text-align: center
+    left: 10%
+    top: 0
+    padding: 0
+    margin: 0 0 0 -28px
+
+@media screen and (max-width: 756px)
```
Comment (KaelWD) [inline src/stylus/components/_timeline.styl]: use the breakpoint variables if possible

File: test/index.js [hunk 7]
```diff
@@ -63,98 +63,95 @@ export function rafPolyfill(w) {
...
-  if (!w.cancelAnimationFrame)  w.cancelAnimationFrame  = cancelRaf;
+  if (!w.requestAnimationFrame) w.requestAnimationFrame = raf
+  if (!w.cancelAnimationFrame) w.cancelAnimationFrame = cancelRaf
 }
 
-export function touch(element) {
+export function touch (element) {
   const createTrigger = eventName => (clientX, clientY) => {
     const touches = [{ clientX, clientY }]
     element.trigger(eventName, ({ touches, changedTouches: touches }))
```
Comment (KaelWD) [inline test/index.js]: Don't change unrelated files

File: src/components/VTimeline/VTimelineItem.js [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+      return this.addTextColorClassChecks(classes, circleOutlineColor)
+    },
+    lineStyles () {
+      const lineSize = parseInt(this.lineSizeParent)
+      const calcMargin = (lineSize - 1) * 3 / 100 + 0.45
+      const width = lineSize > 12 ? `12px` : lineSize < 0 ? `0px` : lineSize
+      const margin = lineSize > 12 ? `-0.78em` : lineSize < 0 ? `-0.42em` : calcMargin
```
Comment (KaelWD) [inline src/components/VTimeline/VTimelineItem.js]: Magic numbers should either be constants or commented

File: .gitignore [hunk 9]
```diff
@@ -18,3 +18,4 @@ dev/*css
 /dist
 /release
 .env
+yarn-error.log
```
Comment (johnleider) [inline .gitignore]: unrelated change, please remove and rebase from dev.

File: src/components/VBtn/VBtn.js [hunk 10]
```diff
@@ -110,7 +110,8 @@ export default {
         children.push(this.$createElement(VProgressCircular, {
           props: {
             indeterminate: true,
-            size: 26
+            size: 23,
```
Comment (johnleider) [inline src/components/VBtn/VBtn.js]: this is already in dev, rebase your changes to only include relevant commits

**Comment by nekosaur** [comment]
I haven't looked at the actual code yet, but here's a couple of comments on the design and use of the component.

- In my opinion `timeline-items` should be vertically centered on their icons, as seen [here](http://rp.js.org/angular-timeline/example/index.html) and [here](https://codepen.io/piitaya/pen/vLappz). Right now they feel too much disconnected from each other.
- There needs to be a slot for putting text/content on the opposite side of the main item
- There needs to be an option for alternating items left and right down the line. Either on timeline or allowing individual items to be flipped.
- `timeline-item` needs an `icon-color` prop.
- `circle-outline-size` prop currently has no effect.
- And on that note (and I might be alone in this), but I feel like there's a little bit of a props overload currently. They're too many and/or too granular, though I'm not sure what if anything could be done about it.

**Comment by jacekkarczmarczyk** [comment]
Something went wrong with the last commit, VBtn, VBtn.spec.js.snap and returnable.js shouldn't be changed in this PR

**Comment by johnleider** [comment]
Closing and opening a branch on the repository, continue dev/conversation there please: #3912
